liblwgeom/lwgeom_topo.c

```diff
@@ -2972,7 +2972,7 @@ _lwt_FindNextRingEdge(const POINTARRAY *ring, int from,
     POINTARRAY *epa = edge->points;
     POINT2D p2, pt;
     int match = 0;
-    int j;
+    int32_t j;
```

Comment:
It does some arithmetic that could lead to UB with uint:
```for ( j=epa->npoints-2; j>=0; --j )```
Since I didn't see no guarantee in the code that epa->npoints is at least 2 I went the easy way. The function would need to be reworked to properly use uint32_t instead.

Reply:
Can you `assert(epa->npoints > 2)` then so it fails assertion and not mysteriously?

Reply:
I'll add an `if` statement since that edge can be safely ignored.